app/scripts/controllers/overview.js

```diff
@@ -188,7 +188,48 @@ angular.module('openshiftConsole')
       $scope.childServicesByParent[parentName].push(child);
     };
 
+    // Assign each service a score for sorting. Services with routes or app
+    // labels are considered more important.
+    var scoreService = function(service) {
+      var score = 0;
+      var name = _.get(service, 'metadata.name', '');
+      var routes = _.get($scope, ['routesByService', name], []);
+      var childServices = _.get($scope, ['childServicesByParent', name], []);
+
+      if (!_.isEmpty(routes)) {
+        score += 5;
+      }
+
+      if (_.has(service, 'metadata.labels.app')) {
+        score += 3;
+      }
+
+      if (!_.isEmpty(childServices)) {
+        score += 1;
+      }
+
+      if (ServicesService.isInfrastructure(service)) {
+        score -= 5;
```

Comment:
i think this is fine for now for this pr but we may want to look at moving
them out later.  we need to decide how important infra stuff is relative to
other edge case stuff.

On Mon, Jun 20, 2016 at 2:43 PM, Sam Padgett <notifications@github.com>
wrote:

> In app/scripts/controllers/overview.js
> <https://github.com/openshift/origin-web-console/pull/72#discussion_r67743702>
> :
>
> > +      var childServices = _.get($scope, ['childServicesByParent', name], []);
> > +
> > +      if (!_.isEmpty(routes)) {
> > +        score += 5;
> > +      }
> > +
> > +      if (_.has(service, 'metadata.labels.app')) {
> > +        score += 3;
> > +      }
> > +
> > +      if (!_.isEmpty(childServices)) {
> > +        score += 1;
> > +      }
> > +
> > +      if (ServicesService.isInfrastructure(service)) {
> > +        score -= 5;
>
> Yeah, you would need 2 extra ng-repeats, one for infrastracuture services
> with app label or routes and another for services without, which would make
> 4 total. I was hesitant to repeat the logic.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-web-console/pull/72/files/2726128c9faa0c3b093ecf85817b4871b2c0a8d7#r67743702>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABZk7bRKEK_Ltfs0lORyzBhaB8IZRJBPks5qNt9FgaJpZM4I4sM6>
> .
>